utils/compare/graph.py

```diff
@@ -0,0 +1,306 @@
+
+        test = metric_of_interest
+        value = row[metric_of_interest]
+        x.append(value/divider)
+        
+        y.append(test)
+    #print('previous x ' + str(x))
+    old = {'y' : x,
+        'x' : y,
+        'name' : 'Previous results ' + str(metric_of_interest),
+        'type' : 'bar',
+        'orientation' : 'v'}
+    x=[]
+    y=[]
+    for n, row in current_data.iterrows():
+        test = metric_of_interest
+        value = row[metric_of_interest]
+        x.append(value/divider)
+        y.append(test)
+    #print('final x ' + str(x))
+    new = {'y' : x,
+        'x' : y,
+        'name' : 'Current results ' + str(metric_of_interest),
+        'type' : 'bar',
+        'orientation' : 'v'}
+    metrics = []
+    metrics.append(old)
+    metrics.append(new)
+
+    return metrics
+
+def pd_merge(previous_data, current_data, columns):
+    pd_merged = current_data.merge(previous_data, how='outer', on=columns[:-1])
+    return pd_merged
+
+def find_result(row, tolerancy): 
+
+    if np.isinf(row['difference']): 
+        return "Pass"
+    # Fail if value is a more negative than tolerancy
+    # High positive valus mean new uuid is better
+    if row['difference'] < 0 and tolerancy > row['difference']: 
+        return 'Fail'
+    else: 
+        return "Pass"
+
+
+def tolerancy(all_data_points, tolerancy_percent): 
+    print('tolerancy ' + str(all_data_points))
+    all_data_points.pad(inplace=True)
+
+    result = all_data_points.iloc[:,[-2,-1]].pct_change(axis=1).iloc[:,1] * 100
+
+    all_with_result = all_data_points.assign(difference=result)
+
+    all_with_result['result'] = all_with_result.apply(lambda row: find_result(row, tolerancy_percent), axis=1)
+
+    return all_with_result
+
+def process_all_data(metric_of_interest, find_metrics,uuid, ids, data_func, index, divider, additional_columns): 
+    # Process all
+
+    columns = []
+    for k in find_metrics.keys(): 
+        columns.append(k)
+    for column_name in additional_columns: 
+        columns.append(column_name)
+    columns.append(metric_of_interest)
+    # Get all previous run data
+    oData = getResults(uuid,ids,index, find_metrics)
+    oMetrics = processData(oData, columns, data_func)
+    
+    oMetrics = oMetrics.reset_index()
+    oMetrics[metric_of_interest] = oMetrics[metric_of_interest] /divider
+
+    #Current data metrics 
+    cData = getResults(uuid,[uuid],index, find_metrics)
+    nMetrics = processData(cData, columns, data_func)
+    nMetrics = nMetrics.reset_index()
+    nMetrics[metric_of_interest] = nMetrics[metric_of_interest] /divider
+    return oMetrics, nMetrics, columns
+
+def generate_pass_fail(oMetrics, nMetrics, columns, tolerancy_num):
+
+    all_merged = pd_merge(oMetrics, nMetrics, columns)
+    print('all merged' + str(all_merged))
+    if not all_merged.empty:
+        all_result_tolerancy = tolerancy(all_merged, tolerancy_num)
+        return all_result_tolerancy
+    return []
+
+def jobSummary(uuids: list):
```

Comment:
just now realized that, this function can also be removed.

Reply:
I think a lot of these are from the graph-api in the dashboard work... 

Preferably, we should create a python package to share these functions, and reuse them across our projects.

Reply:
Expect for the `cpt-dashboard` what is the alternative place that we are planning to visualize these graphs?

Reply:
Honestly, no where to my knowledge. The idea was to share how we build the dataframes for the trending graphs and for the comparisons.

Reply:
IMHO, lets wait for at-least on more tool to have the same use-case of displaying those graphs. Until then its just dead code and can be removed.